source/slides/02_linux_basics.rst

```diff
@@ -0,0 +1,507 @@
+
+``$ git init``
+
+.. note:: This initializes a git repo. Use `man git-init` for more info.
+
+``$ git add <filename>``
+
+.. note:: This puts <filename> into the staging area. It isn't committed yet.
+    Use ``git diff`` to see what changes aren't yet in staging.
+
+``$ git commit -m "I did a thing!"``
+
+.. note:: This actually makes the commit. Use ``git status`` to see what's in
+    staging but not yet committed. Use ``git show`` or ``git log`` to see
+    recent commits.
+
+* Undo things?
+  the `git book <http://git-scm.com/book/en/Git-Basics-Undoing-Things>`_ explains
+  well
+
+* Did I remember to commit that?::
+    
+    $ git status
+
+* What commits have I made lately?::
+
+    $ git log
+
+What Not To Do
+--------------
+
+* Don't delete the .git files
+
+.. note:: If you kill them, git loses its memory :(
+
+* Avoid redundant copies of the same work in one revision
+* Don't make "oops, undoing that" commits.
+    * Use git commit --amend or git revert
+
+.. note:: Amending is fine as long as you haven't pushed yet. It's generally a
+    bad idea to amend or rebase work that you've already shared with others,
+    unless you really know what you're doing.
+
+* Don't wait too long between commits
+    * You can squash them all together later
+
+.. note:: Commit every time you think you might want to return to the current 
+    state. You can revert back to any previous commit, but there is no way to
+    magically add a commit in where you forgot to make one.
+
+* Don't commit secrets...
+
+* If you do commit secrets, `reset them`_
+
+.. _reset them: https://jordan-wright.github.io/blog/2014/12/30/why-deleting-sensitive-information-from-github-doesnt-save-you/
+
+.. note:: Yes, there are ways to sort of take them down off of GitHub, but
+    somebody might have cloned your repo while it had the secrets in. Once
+    someone has a piece of information, you can't just take it away.
+
+Daily workflow
+--------------
+
+.. figure:: ../_static/gitflow.png
+    :scale: 75%
+    :align: right
+
+Pull -> Work -> Add changes -> Commit -> Push
+
+Larger projects have more complex workflows
+
+.. note:: The picture is of the Git Flow branching model, and you'll probably
+    see it every single time anyone explains Git branching and merging to you.
+
+Openstack
+=========
+
+You need:
+
+* Your laptop
+* SSH Key (we can generate one while setting up)
+* SSH (Windows users should use `putty`_
+* Your ONID username and ID #
+
+.. _putty: http://www.chiark.greenend.org.uk/~sgtatham/putty/download.html
+
+Setting Up
+----------
+
+.. figure:: ../_static/openstack_login.png
+
+UserName: ONID username
+Password: ID # (change this)
+
+Setting Up
+----------
+
+.. figure:: ../_static/openstack_change_password.png
+
+If you don't change your password, an automated script will be emailing you!
+
+Setting Up
+----------
+
+.. figure:: ../_static/openstack_overview.png
+
+Setting Up
+----------
+
+.. figure:: ../_static/openstack_security.png
```

Comment:
You might use the width arg here with a percentage so that it fits within the slide.